src/wallet/rpcdump.cpp

```diff
@@ -1161,3 +1161,73 @@ UniValue importmulti(const JSONRPCRequest& mainRequest)
 
     return response;
 }
+
+UniValue rescanblockchain(const JSONRPCRequest& request)
+{
+    CWallet * const pwallet = GetWalletForJSONRPCRequest(request);
+    if (!EnsureWalletIsAvailable(pwallet, request.fHelp)) {
+        return NullUniValue;
+    }
+
+    if (request.fHelp || request.params.size() > 2) {
+        throw std::runtime_error(
+            "rescanblockchain (\"startheight\") (\"stopheight\")\n"
+            "\nRescan the local blockchain for wallet related transactions.\n"
+            "\nArguments:\n"
+            "1. \"startheight\"    (number, optional) blockheight where the rescan should start\n"
+            "2. \"stopheight\"     (number, optional) blockheight where the rescan should stop\n"
+            "\nExamples:\n"
+            + HelpExampleCli("rescanblockchain", "\"100000 120000\"")
+            + HelpExampleRpc("rescanblockchain", "\"100000 120000\"")
+            );
+    }
+
+    LOCK2(cs_main, pwallet->cs_wallet);
+
+    CBlockIndex *pindexStart = nullptr;
+    CBlockIndex *pindexStop = nullptr;
+    if (!request.params[0].isNull()) {
+        pindexStart = chainActive[request.params[0].get_int()];
+        if (!pindexStart) {
+            throw JSONRPCError(RPC_WALLET_ERROR, "Invalid startheight");
+        }
+    }
+    if (!pindexStart) {
+        pindexStart = chainActive.Genesis();
+    }
+
+    if (!request.params[1].isNull()) {
+        pindexStop = chainActive[request.params[1].get_int()];
+        if (!pindexStop || pindexStop->nHeight < pindexStart->nHeight) {
+            throw JSONRPCError(RPC_WALLET_ERROR, "Invalid stopheight");
```

Comment:
If the heights are inverted, they should be flipped. 100..90 is no less rational than 90..100. This was in 79a11731f3, but seems to have disappeared silently in a rebase?

Reply:
@luke-jr - do you have a particular use-case for inverting the block heights? If not, I think this is fine as is. We should err towards being more strict in the interface to keep the code simpler.

Reply:
Already discussed [here](https://github.com/bitcoin/bitcoin/pull/7061#discussion_r132807647).

There's no reason to have an arbitrary restriction that doesn't make sense and is undocumented, especially when we can trivially just do the right thing.

Reply:
'The right thing' is a value judgement. In my opinion, the right thing is to keep APIs as tightly defined as possible in order to keep code simple and remove corner cases.

Of course, if there's a legitimate use case, we should allow both, but I don't see it.

Reply:
The legitimate use case is to rescan the blockchain... There's no reason why one direction is to be preferred over the other for these arguments.